### PR TITLE
Fix issue #1264 incompatible EXTCODEHASH implementation, with hardfork

### PR DESCRIPTION
**Pull request by nicops**
Creating this as draft for review since it has placeholders for hardfork activation name and RSKIP names as well.

 The bulk of the testing is done via DSL, which allowed for testing behavior with and without MutableTrieCache effects and pre and post activation of the hardfork

### CODE REVIEW

File: rskj-core/src/main/java/org/ethereum/db/MutableRepository.java [hunk 1]
```diff
@@ -158,21 +159,47 @@ public synchronized int getCodeLength(RskAddress addr) {
...
 
         byte[] key = trieKeyMapper.getCodeKey(addr);
-        return mutableTrie.getValueHash(key);
-    }
+        Keccak256 valueHash = mutableTrie.getValueHash(key);
+
+        if (valueHash == null) {
+            //This is the non standard implementation we had pre RskipEXTCODEHASH implementation
+            //and thus me must honor it.
+            return Keccak256.ZERO_HASH;
```
[the remark this answers is not in the document]
Comment (nicops) [reply rskj-core/src/main/java/org/ethereum/db/MutableRepository.java:178]: This is the method that implements the previous behavior

**Comment by nicops** [comment]
@julianlen re: `May be you should upload the contracts used for dsl tests and write a comment on the dsl file pointing to that contract.` the contracts are in the top comment of each test

File: rskj-core/src/test/java/co/rsk/vm/ExtCodeHashDslTest.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2018 RSK Labs Ltd.
```
Comment (Vovchyk) [inline rskj-core/src/test/java/co/rsk/vm/ExtCodeHashDslTest.java:3]: `2020` ? J

File: rskj-core/src/main/java/co/rsk/db/MutableTrieCache.java [hunk 3]
```diff
@@ -235,7 +235,7 @@ public Uint24 getValueLength(byte[] key) {
 
     @Override
     public Keccak256 getValueHash(byte[] key) {
```
[the remark this answers is not in the document]
Comment (nicops) [reply rskj-core/src/main/java/co/rsk/db/MutableTrieCache.java]: Changed it to Optional instead

File: rskj-core/src/main/java/co/rsk/db/MutableTrieImpl.java [hunk 4]
```diff
@@ -89,9 +89,8 @@ public Uint24 getValueLength(byte[] key) {
     @Override
     public Keccak256 getValueHash(byte[] key) {
```
[the remark this answers is not in the document]
Comment (nicops) [reply rskj-core/src/main/java/co/rsk/db/MutableTrieImpl.java]: Same, using Optional now

File: rskj-core/src/main/resources/reference.conf [hunk 5]
```diff
@@ -37,7 +37,7 @@ blockchain = {
             rskip152 = papyrus200
             rskip156 = papyrus200
             rskipUMM = papyrus200
-            rskip169 = hop
+            rskip169 = iris
```
Comment (patogallaiovlabs) [inline rskj-core/src/main/resources/reference.conf:40]: iris300?

**Comment by patogallaiovlabs** [comment]
pipeline:run

**Comment by patogallaiovlabs** [comment]
pipeline:run

**Review by ajlopezrsk — APPROVED** [review]

**Pull request merged**